### ISSUE TITLE
Tooltips aren't showing up if the top-level data prop is an empty array

### ISSUE THREAD

**Issue by charpeni** [recharts/recharts#3987]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/998w87?file=%2Fsrc%2FApp.tsx%3A21%2C1-22%2C1)

### Steps to reproduce

A regression was introduced in [`2.9.3`](https://github.com/recharts/recharts/compare/v2.9.2...v2.9.3), where tooltips aren't showing up if the top-level `data` is an empty array.

See the reproduction link above.

The following works perfectly (meaning that tooltips are showing up) on `2.9.2`, but stopped working starting from `2.9.3`:

```tsx
<ComposedChart data={[]}>
  [...]
  <Tooltip />
  <Bar data={pageData} dataKey="uv" />
</ComposedChart>
```

### What is expected?

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/recharts-tooltip-issue-repro-2-9-0-np2qcc?file=%2Fsrc%2FApp.tsx%3A21%2C1-22%2C1)

<img width="634" alt="Screenshot 2023-11-21 at 1 53 07 PM" src="https://github.com/recharts/recharts/assets/7189823/3ac235c0-c812-49cd-a324-0257bf843bcb">

I would expect tooltips to appear even though the top-level `data` is empty, like it used to be the case previously.

### What is actually happening?

See the reproduction link above.

<img width="630" alt="Screenshot 2023-11-21 at 1 53 41 PM" src="https://github.com/recharts/recharts/assets/7189823/aa19e870-7ad5-45a1-a5d3-44bfb099c103">

When upgrading from `2.9.2` to `2.9.3` (or newer), tooltips aren't showing up anymore.

I investigated a little bit, and the reason is that `payload` contained within the tooltip is now an empty array and doesn't contain other data keys.

This is only happening with an empty array:

```tsx
// Tooltips aren't showing up
<ComposedChart data={[]} />

// OK
<ComposedChart />
<ComposedChart data={['keep me']} />
```

As a workaround, we could omit `data` instead of passing an empty array, but in that case, it will be leading to another issue: https://github.com/recharts/recharts/issues/3985.

The only workaround remaining that should satisfy all usages would be to pass a non-empty array with some junk data, e.g., `data={['keep me']}`, but I would be really concerned that in the next version, we end up with that junk data somewhere else impacting our chart.

| Environment | Info |
|---|---|
| Recharts | v2.10.1 |
| React | 18.2.0 |
| System | macOS Sonoma 14.0 |
| Browser | Chrome 119.0.6045.159 |

**Comment by ckifer** [comment]
should be fixed with #3988

will do a patch release at some point soon I think, thanks!

**Comment by ckifer** [comment]
fixed in 2.10.2
